src/Shouldly/ShouldlyExtensionMethods/ShouldBeTypeTestExtensions.cs

```diff
@@ -9,7 +9,7 @@ public static partial class ShouldBeTestExtensions
         public static T? ShouldBeAssignableTo<T>(this object? actual, string? customMessage = null)
         {
             ShouldBeAssignableTo(actual, typeof(T), customMessage);
-            return (T)actual;
+            return (T?)actual;
```

Comment:
Good catch. Was there a warning for this? If not, the called ShouldBeAssignableTo overload is improperly annotated. If so, I wonder why I didn't see it.